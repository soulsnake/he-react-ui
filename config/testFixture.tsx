import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';
import toJSON from 'enzyme-to-json';

interface Fixture {
  component: React.ComponentType<any>;
  name?: string;
}

interface EnzymeElement {
  node: {
    nodeType: string;
    props: {
      children?: null;
    };
  };
  children: null | (EnzymeElement | null)[];
}

function sanitiseSubtree(
  element: EnzymeElement | null,
  compositeSeenPreviously: boolean,
): EnzymeElement | null {
  if (!element) return element;
  const { node, children } = element;
  if (!node) return element;
  const { nodeType, props } = node;
  const isComposite = nodeType !== 'host';
  const hasChildrenProp = Boolean(props.children);
  const shouldIncludeChildren =
    !isComposite || !compositeSeenPreviously || hasChildrenProp;
  const compositeSeen = compositeSeenPreviously || isComposite;
  const cleanedChildren: (EnzymeElement | null)[] | null = shouldIncludeChildren
    ? children &&
      children.map((it: EnzymeElement | null) =>
        sanitiseSubtree(it, compositeSeen),
      )
    : [];

  return {
    ...element,
    children: cleanedChildren,
  };
}

function shallowSerialise(enzymeTree: any): EnzymeElement | null {
  return sanitiseSubtree((toJSON(enzymeTree) as any) as EnzymeElement, false);
}

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, unmount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;

  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => {
    mount();
    try {
      expect(shallowSerialise(getWrapper())).toMatchSnapshot();
    } finally {
      unmount();
    }
  });
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
