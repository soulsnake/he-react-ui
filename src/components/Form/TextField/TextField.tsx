/**
 *
 * TextField
 *
 */

import classnames from 'classnames';
import React from 'react';
import { returnNull } from '../../../util';
import Cross from '../../Icon/Cross';
import Help from '../../Icon/Help';
import Tick from '../../Icon/Tick';
import Popover from '../../Popover';
import style from './TextField.scss';

enum Markers {
  none = 'none',
  tickCross = 'tick-cross',
  info = 'info',
  verify = 'verify',
}

type Props = {
  className?: string;
  description?: string;
  disabled?: boolean;
  error?: string;
  helper?: any;
  id: string;
  inline?: boolean;
  isValid?: boolean;
  isVerified?: boolean;
  label?: Object | string;
  marker?: Markers;
  markerTooltip?: any;
  name: string;
  onBlur: Function;
  onChange: Function;
  onFocus: Function;
  password?: boolean;
  small?: boolean;
  onMarkerClick: () => any;
  value?: string;
  autoComplete?: string;
};

class TextField extends React.Component<Props, any> {
  static defaultProps = {
    disabled: false,
    inline: false,
    marker: 'none',
    markerTooltip: null,
    value: '',
    isValid: true,
    isVerified: false,
    onBlur: returnNull,
    onChange: returnNull,
    onFocus: returnNull,
    onMarkerClick: returnNull,
  };

  state = {
    focused: false,
  };

  handleFocus = (e: React.SyntheticEvent<HTMLInputElement>) => {
    const event = {
      value: (e.currentTarget as any).value,
      props: this.props,
    };

    this.setState({
      focused: true,
    });
    this.props.onFocus(event);
  };

  handleBlur = (e: React.SyntheticEvent<HTMLInputElement>) => {
    const event = {
      value: (e.currentTarget as any).value,
      props: this.props,
    };

    this.setState({
      focused: false,
    });
    this.props.onBlur(event);
  };

  handleChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
    const event = {
      value: (e.currentTarget as any).value,
      props: this.props,
    };
    this.props.onChange(event);
  };

  renderMarker = () => {
    const {
      value,
      marker,
      markerTooltip,
      isVerified,
      isValid,
      onMarkerClick,
    } = this.props;
    let markerIcon = null;

    switch (marker) {
      case 'tick-cross': {
        markerIcon =
          value !== '' &&
          (isValid ? (
            <Tick className={style.marker} onClick={onMarkerClick} />
          ) : (
            <Cross className={style.marker} onClick={onMarkerClick} />
          ));
        break;
      }
      case 'info': {
        markerIcon = <Help className={classnames(style.helperIcon)} />;

        break;
      }
      case 'verify': {
        markerIcon = (
          <div
            className={classnames(
              style.verifiedTag,
              isVerified ? style.verified : style.unVerified,
            )}
            onClick={onMarkerClick}
          >
            {isVerified ? 'Verified' : 'Verify'}
          </div>
        );
        break;
      }
      default: {
        markerIcon = null;
      }
    }

    return (
      <div className={style.helper}>
        {markerTooltip ? (
          <Popover content={markerTooltip} light preferRight>
            {markerIcon}
          </Popover>
        ) : (
          markerIcon
        )}
      </div>
    );
  };
  render() {
    const {
      className,
      id,
      name,
      label,
      description,
      value,
      disabled,
      error,
      inline,
      marker,
      markerTooltip,
      onBlur,
      onChange,
      onFocus,
      isValid,
      isVerified,
      password,
      small,
      onMarkerClick,
      autoComplete,
      ...restProps
    } = this.props;
    const { focused } = this.state;
    const floating = focused || value !== '';
    const wrapperClasses = classnames(
      style.outer,
      {
        [style.invalid]: !isValid,
        [style.disabled]: disabled,
        [style.inline]: inline,
        [style.focused]: focused,
        [style.hasMarker]: marker !== 'none',
        [style.small]: small,
      },
      className,
    );

    const inputClasses = classnames(style.input, {
      [style.small]: small,
      [style.noLabel]: !label,
    });

    const labelClasses = classnames(style.label, {
      [style.floating]: floating,
      [style.small]: small,
    });

    return (
      <div className={wrapperClasses} {...restProps}>
        <div className={style.block}>
          <input
            className={inputClasses}
            id={id}
            name={name}
            onFocus={this.handleFocus}
            onBlur={this.handleBlur}
            onChange={this.handleChange}
            disabled={disabled}
            value={value}
            type={password ? 'password' : 'text'}
            autoComplete={autoComplete || undefined}
          />

          {this.renderMarker()}

          {label && (
            <label className={labelClasses} htmlFor={id}>
              {label}
            </label>
          )}
        </div>

        {(description || error) && (
          <label htmlFor={id} className={style.description}>
            {error || description}
          </label>
        )}
      </div>
    );
  }
}

export default TextField;
