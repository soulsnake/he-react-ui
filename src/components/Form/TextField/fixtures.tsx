// @flow
import React, { Fragment } from 'react';
import TextField from '.';
import { Heading, Paragraph, Icon } from '../../../';
import style from './TextField.scss';

export default [
  {
    name: 'Default',
    formWrapper: { value: '' },
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
    },
  },

  {
    name: 'small',
    formWrapper: { value: '' },
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      small: true,
    },
  },

  {
    name: 'smallNoLabel',
    formWrapper: { value: '' },
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      small: true,
    },
  },

  {
    name: 'password',
    formWrapper: { value: '' },
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'Password',
      password: true,
    },
  },

  {
    name: 'noLabel',
    formWrapper: { value: '' },
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
    },
  },

  {
    name: 'Disabled',
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      disabled: true,
    },
  },

  {
    name: 'Error',
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      error: "Something isn't right",
      isValid: false,
    },
  },

  {
    name: 'InvalidMarker',
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      error: "Something isn't right",
      marker: 'tick-cross',
      value: 'Bruce',
      isValid: false,
    },
  },

  {
    name: 'ValidMarker',
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      marker: 'tick-cross',
      value: 'Bruce',
    },
  },

  {
    name: 'Value',
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      value: 'Oscar',
    },
  },

  {
    name: 'Info tooltip',
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      marker: 'info',
      markerTooltip: <div>Test HTML</div>,
    },
  },

  {
    name: 'Verified status',
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'Mobile',
      marker: 'verify',
    },
  },

  {
    name: 'UnVerified status',
    component: TextField,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'Mobile',
      marker: 'verify',
      isVerified: false,
      markerTooltip: (
        <Fragment>
          <Icon key="icon" name="Edit" className={style.editIcon} />
          <Heading key="h3" h3>
            Sample Popup
          </Heading>
          <Paragraph key="paragraph" className={style.paragraph}>
            Sample content
          </Paragraph>
        </Fragment>
      ),
    },
  },

  {
    name: 'Autocomplete off',
    component: TextField,
    props: {
      id: 'firstname',
      name: 'firstname',
      label: 'First Name',
      autoComplete: 'off',
    },
  },
];
