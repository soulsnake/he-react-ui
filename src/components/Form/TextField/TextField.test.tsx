import { mount } from 'enzyme';
import React from 'react';
import TextField from './';
import fixtures from './fixtures';

import testFixture from '../../../../config/testFixture';

fixtures.forEach(it => testFixture(it));

describe('TextField', () => {
  it('test interaction', () => {
    const handleOnChange = jest.fn();
    const harness = mount(
      <TextField
        id="demo"
        name="demo"
        label="Label"
        onChange={handleOnChange}
      />,
    );

    harness.find('input').simulate('focus');
    harness.find('input').simulate('blur');
    harness.find('input').simulate('change');
    expect(harness).toMatchSnapshot();
  });
});
