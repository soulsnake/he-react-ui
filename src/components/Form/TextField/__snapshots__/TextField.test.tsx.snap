// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> rendered correctly with Autocomplete off fixture 1`] = `
<TextField
  autoComplete="off"
  disabled={false}
  id="firstname"
  inline={false}
  isValid={true}
  isVerified={false}
  label="First Name"
  marker="none"
  markerTooltip={null}
  name="firstname"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer"
  >
    <div
      className="block"
    >
      <input
        autoComplete="off"
        className="input"
        disabled={false}
        id="firstname"
        name="firstname"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      />
      <label
        className="label"
        htmlFor="firstname"
      >
        First Name
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with Default fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="First Name"
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      />
      <label
        className="label"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with Disabled fixture 1`] = `
<TextField
  disabled={true}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="First Name"
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer disabled"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={true}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      />
      <label
        className="label"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with Error fixture 1`] = `
<TextField
  disabled={false}
  error="Something isn't right"
  id="demo"
  inline={false}
  isValid={false}
  isVerified={false}
  label="First Name"
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer invalid"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      />
      <label
        className="label"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
    <label
      className="description"
      htmlFor="demo"
    >
      Something isn't right
    </label>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with Info tooltip fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="First Name"
  marker="info"
  markerTooltip={
    <div>
      Test HTML
    </div>
  }
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer hasMarker"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      >
        <Popover
          content={
            <div>
              Test HTML
            </div>
          }
          light={true}
          preferRight={true}
        >
          <Tooltip
            animateFill={true}
            animation="shift"
            arrow={false}
            arrowSize="regular"
            className=""
            delay={100}
            disabled={false}
            distance={10}
            duration={375}
            followCursor={false}
            hideDelay={300}
            hideDuration={375}
            hideOnClick={true}
            html={
              <PopoverDisplay
                className={undefined}
                light={true}
                tooltip={undefined}
              >
                <div>
                  Test HTML
                </div>
              </PopoverDisplay>
            }
            inertia={false}
            interactive={true}
            interactiveBorder={16}
            multiple={false}
            offset={0}
            onHidden={[Function]}
            onHide={[Function]}
            onRequestClose={[Function]}
            onShow={[Function]}
            onShown={[Function]}
            popperOptions={Object {}}
            position="bottom-end"
            size="regular"
            sticky={false}
            stickyDuration={200}
            style={Object {}}
            theme="dark"
            touchHold={false}
            trigger="mouseenter"
            unmountHTMLWhenHide={false}
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline",
                }
              }
            >
              <Help
                className="helperIcon"
                fill="currentColor"
                height={32}
                width={32}
              />
            </div>
          </Tooltip>
        </Popover>
      </div>
      <label
        className="label"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with InvalidMarker fixture 1`] = `
<TextField
  disabled={false}
  error="Something isn't right"
  id="demo"
  inline={false}
  isValid={false}
  isVerified={false}
  label="First Name"
  marker="tick-cross"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value="Bruce"
>
  <div
    className="outer invalid hasMarker"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value="Bruce"
      />
      <div
        className="helper"
      >
        <Cross
          className="marker"
          fill="currentColor"
          height={32}
          onClick={[Function]}
          width={32}
        />
      </div>
      <label
        className="label floating"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
    <label
      className="description"
      htmlFor="demo"
    >
      Something isn't right
    </label>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with UnVerified status fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="Mobile"
  marker="verify"
  markerTooltip={
    <UNDEFINED>
      <mockConstructor
        className="editIcon"
        name="Edit"
        shadow={false}
      />
      <mockConstructor
        h1={false}
        h2={false}
        h3={true}
        h4={false}
        nav={false}
      >
        Sample Popup
      </mockConstructor>
      <Paragraph
        className="paragraph"
      >
        Sample content
      </Paragraph>
    </UNDEFINED>
  }
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer hasMarker"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      >
        <Popover
          content={
            <UNDEFINED>
              <mockConstructor
                className="editIcon"
                name="Edit"
                shadow={false}
              />
              <mockConstructor
                h1={false}
                h2={false}
                h3={true}
                h4={false}
                nav={false}
              >
                Sample Popup
              </mockConstructor>
              <Paragraph
                className="paragraph"
              >
                Sample content
              </Paragraph>
            </UNDEFINED>
          }
          light={true}
          preferRight={true}
        >
          <Tooltip
            animateFill={true}
            animation="shift"
            arrow={false}
            arrowSize="regular"
            className=""
            delay={100}
            disabled={false}
            distance={10}
            duration={375}
            followCursor={false}
            hideDelay={300}
            hideDuration={375}
            hideOnClick={true}
            html={
              <PopoverDisplay
                className={undefined}
                light={true}
                tooltip={undefined}
              >
                <UNDEFINED>
                  <mockConstructor
                    className="editIcon"
                    name="Edit"
                    shadow={false}
                  />
                  <mockConstructor
                    h1={false}
                    h2={false}
                    h3={true}
                    h4={false}
                    nav={false}
                  >
                    Sample Popup
                  </mockConstructor>
                  <Paragraph
                    className="paragraph"
                  >
                    Sample content
                  </Paragraph>
                </UNDEFINED>
              </PopoverDisplay>
            }
            inertia={false}
            interactive={true}
            interactiveBorder={16}
            multiple={false}
            offset={0}
            onHidden={[Function]}
            onHide={[Function]}
            onRequestClose={[Function]}
            onShow={[Function]}
            onShown={[Function]}
            popperOptions={Object {}}
            position="bottom-end"
            size="regular"
            sticky={false}
            stickyDuration={200}
            style={Object {}}
            theme="dark"
            touchHold={false}
            trigger="mouseenter"
            unmountHTMLWhenHide={false}
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline",
                }
              }
            >
              <div
                className="verifiedTag unVerified"
                onClick={[Function]}
              >
                Verify
              </div>
            </div>
          </Tooltip>
        </Popover>
      </div>
      <label
        className="label"
        htmlFor="demo"
      >
        Mobile
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with ValidMarker fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="First Name"
  marker="tick-cross"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value="Bruce"
>
  <div
    className="outer hasMarker"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value="Bruce"
      />
      <div
        className="helper"
      >
        <Tick
          className="marker"
          fill="currentColor"
          height={32}
          onClick={[Function]}
          width={32}
        />
      </div>
      <label
        className="label floating"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with Value fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="First Name"
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value="Oscar"
>
  <div
    className="outer"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value="Oscar"
      />
      <div
        className="helper"
      />
      <label
        className="label floating"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with Verified status fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="Mobile"
  marker="verify"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer hasMarker"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      >
        <div
          className="verifiedTag unVerified"
          onClick={[Function]}
        >
          Verify
        </div>
      </div>
      <label
        className="label"
        htmlFor="demo"
      >
        Mobile
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with noLabel fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer"
  >
    <div
      className="block"
    >
      <input
        className="input noLabel"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      />
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with password fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="Password"
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  password={true}
  value=""
>
  <div
    className="outer"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="password"
        value=""
      />
      <div
        className="helper"
      />
      <label
        className="label"
        htmlFor="demo"
      >
        Password
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with small fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="First Name"
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  small={true}
  value=""
>
  <div
    className="outer small"
  >
    <div
      className="block"
    >
      <input
        className="input small"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      />
      <label
        className="label small"
        htmlFor="demo"
      >
        First Name
      </label>
    </div>
  </div>
</TextField>
`;

exports[`<TextField /> rendered correctly with smallNoLabel fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  small={true}
  value=""
>
  <div
    className="outer small"
  >
    <div
      className="block"
    >
      <input
        className="input small noLabel"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      />
    </div>
  </div>
</TextField>
`;

exports[`TextField test interaction 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="Label"
  marker="none"
  markerTooltip={null}
  name="demo"
  onBlur={[Function]}
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "props": Object {
              "disabled": false,
              "id": "demo",
              "inline": false,
              "isValid": true,
              "isVerified": false,
              "label": "Label",
              "marker": "none",
              "markerTooltip": null,
              "name": "demo",
              "onBlur": [Function],
              "onChange": [MockFunction] {
                "calls": [Circular],
              },
              "onFocus": [Function],
              "onMarkerClick": [Function],
              "value": "",
            },
            "value": "",
          },
        ],
      ],
    }
  }
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      />
      <label
        className="label"
        htmlFor="demo"
      >
        Label
      </label>
    </div>
  </div>
</TextField>
`;
